### PR TITLE
Add school leaderboard scraper script.

### PR DESCRIPTION
**Pull request by NifleySnifley**
It uses a JSON file to specify the schools and leaderboard codes for convenience of updating, specify the path to this json in the `COMPETITION_FILE` environment variable. (add to `.env`)

### CODE REVIEW

File: jobs/pullAocLeaderboards2024.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from pathlib import Path
+
+# Load environment variables from .env.
+load_dotenv()  
+
+YEAR = str(os.getenv('AOC_YEAR'))
+COMPETITION_FILE = os.getenv('COMPETITION_FILE')
+SESSION_TOKEN = os.getenv('AOC_TOKEN')
+
+OUTPUT_FILE_NAME = f"schoolScores{YEAR}.json"
```
Comment (mike-weiner) [inline jobs/pullAocLeaderboards2024.py:16]: Can we change the output file name to be `generatedAocLeaderboard2024.json`? Just want to keep it consistent with previous years' naming convention.

File: .gitignore [hunk 2]
```diff
@@ -40,4 +40,5 @@ yarn-error.log*
 
 # Ignore ALL Typescript files.
 *.tsbuildinfo
-next-env.d.ts
\ No newline at end of file
+next-env.d.ts
+competition.json
```
Comment (mike-weiner) [inline .gitignore:44]: Not sure if we need to add `competition.json` here. This file is going to be kept in `generatedData/` and will probably be called `competitionData2024.json`.

File: jobs/pullAocLeaderboards2024.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import os
+import requests
+import time
+from pathlib import Path
+
+# Load environment variables from .env.
+load_dotenv()  
+
+YEAR = str(os.getenv('AOC_YEAR'))
+COMPETITION_FILE = os.getenv('COMPETITION_FILE')
```
Comment (mike-weiner) [inline jobs/pullAocLeaderboards2024.py:13]:
```suggestion
COMPETITION_FILE_PATH = os.getenv('COMPETITION_FILE')
```

**Review by mike-weiner — CHANGES_REQUESTED** [review]
This looks awesome! Just a couple of small comments.

**Comment by NifleySnifley** [comment]
Thanks for the tweaks, I've updated the naming convention and switched around the `.gitignore`.

**Review by mike-weiner — APPROVED** [review]
/lgtm

🚀 - Looks great! Thanks for doing this!

**Pull request merged**